src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,335 @@
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def exists(self, path: str) -> bool:
+        """Checks if a file/directory path exists.
+
+        Args:
+            path (str): File/directory path
+
+        Returns:
+            bool: If the file/directory  exists or not
+        """
+        try:
+            return self.fs.exists(path)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def ls(self, path: str) -> list[str]:
+        """List the directory path.
+
+        Args:
+            path (str): Directory path
+
+        Returns:
+            List[str]: List of files / directories under the path
+        """
+        try:
+            return self.fs.ls(path)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def rm(self, path: str, recursive: bool = True):
+        """Removes a file or directory mentioned in path.
+
+        Args:
+            path (str): Path to the file / directory
+            recursive (bool): Whether the files and folders nested
+            under path are to be removed or not
+
+        Returns:
+            NA
+        """
+        try:
+            return self.fs.rm(path=path, recursive=recursive)
+        except FileNotFoundError as e:
+            logger.debug(f"Path {path} does not exist.")
+            raise e
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def cp(self, src: str, dest: str, overwrite: bool = True):
+        """Copies files from source(lpath) path to the destination(rpath) path.
+
+        Args:
+            src (str): Path to the source
+            dest (str): Path to the destination
+
+        Returns:
+            NA
+        """
+        try:
+            return self.fs.cp(src, dest, overwrite=overwrite)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def size(self, path: str) -> int:
+        """Get the size of the file specified in path.
+
+        Args:
+            path (str): Path to the file
+
+        Returns:
+            int: Size of the file in bytes
+        """
+        try:
+            file_info = self.fs.info(path)
+            return file_info["size"]
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def modification_time(self, path: str) -> datetime:
+        """Get the last modification time of the file specified in path.
+
+        Args:
+            path (str): Path to the file
+
+        Returns:
+            datetime: Last modified time in datetime
+        """
+        try:
+            file_info = self.fs.info(path)
+            file_mtime = file_info["mtime"]
+            if not isinstance(file_mtime, datetime):
+                file_mtime = datetime.fromtimestamp(file_mtime)
+            return file_mtime
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def mime_type(self, path: str) -> str:
```

Comment:
NIT: @gaya3-zipstack how is this different from the function in `/utils`? Can we aim to retain one of the implementations alone eventually?

Reply:
Idea is to retain one.  Once we have moved everywhere to FileStorage, we should have only this one and remove the other versions.

Reply:
I have added ToDO in other functions.